### PR TITLE
Enable C++14 for RmlUi

### PR DESCRIPTION
**Pull request by slipher**
_No description provided._

---

_Edit by illwieckz._

Needed by:

- https://github.com/Unvanquished/Unvanquished/pull/1715

See also:

- https://github.com/UnvanquishedAssets/unvanquished_src.dpkdir/pull/95

### CODE REVIEW

**Comment by necessarily-equal** [comment]
I'll trust you on that one

File: src/common/System.cpp [hunk 1]
```diff
@@ -391,9 +391,17 @@ void* operator new(size_t n)
...
 	return p;
 }
+
 void operator delete(void* p) NOEXCEPT
 {
 	if (!Sys::processTerminating) {
 		free(p);
 	}
 }
+
```
Comment (bmorel) [inline src/common/System.cpp:401]: makes me thinking about 3 (not related to C++14) things:

1. there is no new[] / delete[] operators
2. builds don't pass when using USE_ADDRESS_SANITIZER

Not really the point of the PR, though.

**Comment by bmorel** [comment]
LGTM. I don't know the relations with NaCl, so like necessarily-equal I'll trust you on that point.

File: src/common/System.cpp (hunk 1 above)
Comment (necessarily-equal) [reply src/common/System.cpp:401]: This makes me think that for ASAN we could simply encapsulate the whole block like [this](https://clang.llvm.org/docs/AddressSanitizer.html#conditional-compilation-with-has-feature-address-sanitizer):
```C++
#if !defined(__has_feature) || !__has_feature(address_sanitizer)
// code that builds only when AddressSanitizer is disabled
#endif
```

Comment (slipher) [reply src/common/System.cpp:401]: Yeah that's what I did on my inactive ASAN branch. https://github.com/slipher/Daemon/commit/237ff1b764d4eb56ef6ac64888d881992971575f

**Review by illwieckz — APPROVED** [review]

**Comment by illwieckz** [comment]
_(I just reverted the branch to its original state after `master` was mistakenly merged into it by @bmorel)._

**Comment by illwieckz** [comment]
I rebased on current `master`.

**Comment by illwieckz** [comment]
Well no, I reverted to before the rebase, something very wrong happened while rebasing. @slipher I'll let you fix the conflict.

**Comment by illwieckz** [comment]
The code works for me but it breaks as soon as I rebase this branch to master.

I also noticed that on Unvanquished side the PR targets `0.53.0/sync` branch, but this one targets `master`.

Otherwise, everything looks good to me, it seems to work as expected.

**Pull request merged**